lnwallet/channel.go

```diff
@@ -1958,6 +1983,19 @@ func (lc *LightningChannel) restoreStateLogs(
 	for i := range remoteCommitment.outgoingHTLCs {
 		htlc := remoteCommitment.outgoingHTLCs[i]
 
+		// If the pending remote commit exists, do not add outgoing HTLCs that
+		// exist on the remote commit, but not the pending remote. We use the
+		// htlc's LogIndex to query the map for existence.
+		//
+		// NOTE: The nil-check is necessary as otherwise HTLC's won't be found
+		// in the map, and we won't add them to the update logs. This would
+		// lead to channel force closure.
+		if pendingRemoteCommit != nil {
+			if _, ok := pendingRemoteOutgoingIndices[htlc.LogIndex]; !ok {
```

Comment:
So I get that the initial issue was that the settle wasn't deleted properly, so then when we go to sign a new signature we'll be covering the wrong commitment, but is this new logic to filter out when restoring just to allow those in this position to resume properly under certain cases?

Reply:
We're A here:
```
A                      B
    ----add---->
    ----sig---->
    <---rev-----
    <---sig-----
    ----rev---->

    <--settle----
    <---sig------
    ----rev----->  (A stores the settle to sign for it)
    ----sig----->  (A deletes the settle here as A has signed for it)
```
If A restarts, A has two commitments for B, the remote commitment and the pending remote commitment.  If A restores the original add into the update logs, there is no corresponding settle as it's been deleted.  So, if A has an outgoing HTLC on the remote commitment and no outgoing HTLC on the pending remote commitment based on the `outgoingHTLCs` struct (which includes dust htlcs), we won't restore the add.